sources.go

```diff
@@ -102,3 +103,47 @@ func (s pstateSource) Discover() ([]string, error) {
 
 	return features, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Network Source
+
+// Implements main.FeatureSource.
+type networkSource struct{}
+
+func (s networkSource) Name() string { return "netid" }
+func (s networkSource) Discover() ([]string, error) {
+	features := []string{}
+	var total_num_vfs int = 0
+	// sysfs of the node mounted as /hostsys inside the pod
+	netInterfaces, err := ioutil.ReadDir("/hostsys/class/net") //does not return . and ..
+	if err != nil {
+		return nil, fmt.Errorf("can't obtain the network interfaces: %s", err.Error())
+	}
+	for _, netInterface := range netInterfaces {
+		//forming the file name : /hostsys/class/net/<network interface>/device/sriov_numvfs
+		filename := "/hostsys/class/net/" + netInterface.Name() + "/device/sriov_numvfs"
+		bytes_received, err := ioutil.ReadFile(filename)
+		if err != nil {
+			// interface does not support SRIOV
+			continue // proceeding to next iteration for next network interface
+		}
+		//removing white spaces
+		num := bytes.TrimSpace(bytes_received)
+		//checking if 0 bytes read
+		if l := len(bytes_received); l == 0 {
+			// Zero bytes read for this file
+			continue // proceeding to next iteration for next network interface
+		}
+		//converting bytes to integer and checking for error
+		n, err := strconv.Atoi(string(num))
```

Comment:
Can we do this here instead?

```go
n, err := strconv.Atoi(string(num)) 
if err != nil && n > 0 {
 features = append(features, "SRIOV")
 break
}
```

This way we need not check the other interfaces and also avoid counting the number of VFs.